modules/logging/filters.go

```diff
@@ -185,15 +187,66 @@ func (m IPMaskFilter) Filter(in zapcore.Field) zapcore.Field {
 	return in
 }
 
+// RegexpFilter is a Caddy log field filter that
+// replaces the field matching the provided regexp with the indicated string.
+type RegexpFilter struct {
+	RawRegexp string `json:"regexp,omitempty"`
+	Value     string `json:"value,omitempty"`
+	regexp    *regexp.Regexp
+}
```

Comment:
Same as other PRs, could use a slightly expanded godoc comment, plus on the struct fields as well